Show hierarchical view in project list

rbt-mm opened the pull request:
Added a switch in the project list to either display a hierarchical view, which shows the (already existing) multi-level parent-child-relationships between projects, or to display the currently existing flat view of projects.

Added new select inputs in the dialogs for creating a project or showing its details, which include every possible parent for the current or new project and create a persistent relationship between them on selection.

Signed-off-by: RBickert <rbt@mm-software.com>

rbt-mm committed:
Add parent in project creation and detail view

Display list of every project in `Create Project` and `View
Project` to select a parent project for the new or current project.

Send selected parent in project creation via API call to backend to
create a persistent parent-child-relationship.

Signed-off-by: RBickert <rbt@mm-software.com>

rbt-mm committed:
Add persistent selection of parent to detail view

Send selected parent in project detail view via API call to backend to
create a persistent parent-child-relationship.

Signed-off-by: RBickert <rbt@mm-software.com>

rbt-mm committed:
Add hierarchical view in project list

Added CSwitch in project list to switch between current flat view
(default) and a new hierarchical view (shows root projects and their
children in the detail view of a row).

Project detail view only shows viable projects as selectable parent.

Hierarchical project list shows inactive projects for children and
children are sortable.

Signed-off-by: RBickert <rbt@mm-software.com>

rbt-mm committed:
Merge branch 'rbt-limit-alert-rules-to-project-including-all-existing-or-future-versions' into rbt-show-hierarchical-view-in-project-list

rbt-mm committed:
Merge branch 'rbt-limit-alert-rules-to-project-including-all-existing-or-future-versions' into rbt-show-hierarchical-view-in-project-list

Signed-off-by: RBickert <rbt@mm-software.com>

rbt-mm committed:
Merge remote-tracking branch 'origin/rbt-show-hierarchical-view-in-project-list' into rbt-show-hierarchical-view-in-project-list

Signed-off-by: RBickert <rbt@mm-software.com>

rbt-mm committed:
Fix switching parent to inactive

A parent project cannot be set to inactive if any of his children are
set to active

Signed-off-by: RBickert <rbt@mm-software.com>

rkg-mm commented on src/views/portfolio/projects/ProjectDetailsModal.vue:
```diff
@@ -174,6 +187,50 @@
           this.$toastr.w(this.$t('condition.unsuccessful_action'));
         });
+      },
+      retrieveParents: function() {
```
Das kostet performance - was genau wird hier erreicht?

rbt-mm replied:
Hier werden alle möglichen Parent-Projekte für ein Projekt mit einem API-Call geholt und so gefiltert, dass das jetzige Projekt und alle Kinder des jetzigen Projekts von der Selection ausgeschlossen sind. Zudem wird der Wert des "Select Parent"-Feld bei der Initialisierung auf den Wert des Parent vom Projekt gesetzt.

Die Funktion ist der schon vorhandenen Funktion `retrieveLicences()` in `ProjectCreateProjectModal.vue` nachempfunden:
https://github.com/rbt-mm/frontend/blob/94c10dcc8bc9a8b7da2f4fbf8037355c8c4d0e7f/src/views/portfolio/projects/ProjectCreateProjectModal.vue#L188-L201

rbt-mm committed:
Filter detail view parents in backend

Available parents in a project detail view are now filtered in the
backend instead of the UI

Signed-off-by: RBickert <rbt@mm-software.com>
